File: ios/Midikit.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(Midikit, NSObject)

RCT_EXTERN_METHOD(multiply:(float)a withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(start:(RCTPromiseResolveBlock)resolve withRejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(sendMidiMessage: (nonnull NSString *)n a: (nonnull NSNumber *)a b: (nonnull NSNumber *)b  c: (nonnull NSNumber *)c )
+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/Midikit.swift
import Foundation

@objc(Midikit)
class Midikit: NSObject {

  @objc(multiply:withB:withResolver:withRejecter:)
  func multiply(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
    print("multiply")
    resolve(a*b)
  }
    
  @objc
  func sendMidiMessage(_ n: NSString, a: NSNumber, b: NSNumber, c: NSNumber) {
      let formattedString = "send midi: \(a), \(b), \(c)"
      NSLog("%@", formattedString)
      print("hello world")
    }
}
